hal/watchdog_api.h

```diff
@@ -99,6 +99,10 @@ typedef struct {
      * You can stop the watchdog after it starts without a reset.
      */
     bool disable_watchdog;
+    /**
+     * Accuracy of frequency not calibrated watchdog clock.
+     */
+    uint8_t clock_accuracy;
```

Comment:
This is public API file, is this changes, we need to be more cautious. all the public documents related to this , and we will need to modify all possible drivers, and think about backward compatibility @fkjagodzinski @Tharazi97

Reply:
By changing initial value of clock accuracy to ``uint8_t clock_accuracy = 100;``. We would need to change only the implementation of function ``hal_watchdog_get_platform_features()`` for targets which have their watchdog clock frequency differ very much. 
This change only adds new functionality, so I don't think there would be problems with backward compatibility.

Reply:
Yes we will need to change every ``hal_watchdog_get_platform_features()`` function. I think it will be backward compatibile, because we are only adding one variable to a structure which for now will be used only for this test, but possibly could be more useful later.